### PR TITLE
automl: break up and simplify the dataset tests

### PR DESCRIPTION
**Pull request by nnegrey**

### CODE REVIEW

File: automl/cloud-client/src/test/java/com/example/automl/LanguageEntityExtractionCreateDatasetTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    // Create a random dataset name with a length of 32 characters (max allowed by AutoML)
+    // To prevent name collisions when running tests in multiple java versions at once.
+    // AutoML doesn't allow "-", but accepts "_"
+    String datasetName =
+        String.format("test_%s", UUID.randomUUID().toString().replace("-", "_").substring(0, 26));
+    LanguageEntityExtractionCreateDataset.createDataset(PROJECT_ID, datasetName);
+
+    String got = bout.toString();
+    assertThat(got).contains("Dataset id:");
+    datasetId = got.split("Dataset id: ")[1].split("\n")[0];
```
Comment (averikitsch) [inline automl/cloud-client/src/test/java/com/example/automl/LanguageEntityExtractionCreateDatasetTest.java]: Is there a purpose for getting the datasetId?

File: automl/cloud-client/src/test/java/com/example/automl/ListDatasetsTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  @After
+  public void tearDown() {
+    System.setOut(null);
+  }
+
+  @Test
+  public void testListDataset() throws IOException {
+    // Act
```
Comment (averikitsch) [inline automl/cloud-client/src/test/java/com/example/automl/ListDatasetsTest.java]: nit: remove or add more specific comment

File: automl/cloud-client/src/test/java/com/example/automl/VisionClassificationCreateDatasetTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2019 Google LLC
```
Comment (averikitsch) [inline automl/cloud-client/src/test/java/com/example/automl/VisionClassificationCreateDatasetTest.java]: 2020?

File: automl/cloud-client/pom.xml [hunk 4]
```diff
@@ -27,30 +27,17 @@
...
-    <maven.compiler.source>1.11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
+    <maven.compiler.source>11</maven.compiler.source>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
   </properties>
 
-  <!-- [START automl_java_dependencies] -->
-  <!--  Using libraries-bom to manage versions.
-  See https://github.com/GoogleCloudPlatform/cloud-opensource-java/wiki/The-Google-Cloud-Platform-Libraries-BOM -->
-  <dependencyManagement>
```
[the remark this answers is not in the document]
Comment (nnegrey) [reply automl/cloud-client/pom.xml]: No, not yet. 

It is pushed to java-cloud-bom: https://github.com/googleapis/java-cloud-bom/blob/master/pom.xml#L207

Which is also in cloud-opensource-java: https://github.com/GoogleCloudPlatform/cloud-opensource-java/blob/ca1dede295e2db75fc7a8148fddadbbf102af80c/boms/cloud-oss-bom/pom.xml#L48

But not sure of their release process. But 3.3.0 did not contain it.

File: automl/cloud-client/src/test/java/com/example/automl/LanguageEntityExtractionCreateDatasetTest.java (hunk 1 above)
Comment (nnegrey) [reply automl/cloud-client/src/test/java/com/example/automl/LanguageEntityExtractionCreateDatasetTest.java]: To delete it in the teardown.

**Comment by nnegrey** [comment]
Note: 

Adding ignore to several long running tests that will be fixed in:
1. https://github.com/GoogleCloudPlatform/java-docs-samples/pull/1929
2. https://github.com/GoogleCloudPlatform/java-docs-samples/pull/1928
3. https://github.com/GoogleCloudPlatform/java-docs-samples/pull/1931

Skipped tests:
- Model deployment
- Undeployment
- Batch prediction

**Pull request merged**